### PR TITLE
Feature: docker

### PR DESCRIPTION
**Pull request by jamesjwarren**
Initial dockerfile and compose config.

### CODE REVIEW

**Comment by jamesjwarren** [comment]
@edwardoparearyee the issue you were having should now be resolved. I've configured the app to run with `foreverjs` in production which will ensure it restarts if it crashes. It also provides the option to restart on file changes in development, you can run this locally with `forever -w app.js`.

I've also added image building and jshint tests in circle. I will add some more documentation to the README.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
+#
+# FM-Frontend-API Docker File
+# Builds a docker image to run the FM frontend services
+#
+
+FROM node:slim
+
+RUN apt-get update -y && apt-get install --no-install-recommends -y -q curl python build-essential git ca-certificates
```
Comment (krak3n) [inline Dockerfile]: Do you need python, build-essential?

Comment (krak3n) [inline Dockerfile]: Also do you need curl?

File: .dockerignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.tmp
+node_modules/*
+assets/app/components/*
+assets/node_modules/*
```
Comment (krak3n) [inline .dockerignore:5]: might wanna add `.git` to this

File: Dockerfile (hunk 1 above)
Comment (jamesjwarren) [inline Dockerfile]: Node.js is built with GYP and thus some packages require it when they are installed  https://github.com/joyent/node/wiki/installation#building-prerequisites. I'm not sure that any of the specific packages we are using actually require it though, I'll check. Probably don't need curl, good point.

**Comment by edoparearyee** [comment]
Working for me now :+1:

**Comment by jamesjwarren** [comment]
@krak3n built with an alpine base and optomised deps, gets the total image size down to ~580 mb from ~860 mb

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+RUN mkdir -p /app/.tmp/public && \
+    cp -r dist/* /app/.tmp/public/ && \
+    rm -rf `find . -type f | grep -v '.*\/dist\/.*'`
+
+WORKDIR /app
+
+# Cleanup
+RUN npm uninstall -g bower && \
+    npm cache clean && \
+    apk del make gcc g++ python git
```
Comment (krak3n) [inline Dockerfile]: even though you do this the layers where they are installed still exist and add to the image size. You may as well leave them installed.

Comment (jamesjwarren) [inline Dockerfile]: Ahh I see, wondered why it didn't reduce the size a lot!

**Comment by krak3n** [comment]
I'm still surprised at that large image size. TBSeen DMS has more build dependencies with a much smaller image size.

Can you make sure `.git` is int the `.dockerignore`.

Also once you have rebuilt try a `docker history` on the image, it will show you the layers, what they did and the size of the layer. I'm just curious as to what adds to the size.

**Comment by krak3n** [comment]
Lets Get This Merged!

**Pull request merged**